commit eea5119fa5979c350af5783a8148eacdd4219715
Author: Steve French <stfrench@microsoft.com>
Date:   Tue Jan 28 01:04:23 2025 -0600

    smb3: add support for IAKerb
    
    There are now more servers which advertise support for IAKerb (passthrough
    Kerberos authentication via proxy).  IAKerb is a public extension industry
    standard Kerberos protocol that allows a client without line-of-sight
    to a Domain Controller to authenticate. There can be cases where we
    would fail to mount if the server only advertises the OID for IAKerb
    in SPNEGO/GSSAPI.  Add code to allow us to still upcall to userspace
    in these cases to obtain the Kerberos ticket.
    
    Signed-off-by: Steve French <stfrench@microsoft.com>

diff --git a/fs/smb/client/cifsglob.h b/fs/smb/client/cifsglob.h
index 5ba6b46fe9d1..a68434ad744a 100644
--- a/fs/smb/client/cifsglob.h
+++ b/fs/smb/client/cifsglob.h
@@ -151,6 +151,7 @@ enum securityEnum {
 	NTLMv2,			/* Legacy NTLM auth with NTLMv2 hash */
 	RawNTLMSSP,		/* NTLMSSP without SPNEGO, NTLMv2 hash */
 	Kerberos,		/* Kerberos via SPNEGO */
+	IAKerb,			/* Kerberos proxy */
 };
 
 enum upcall_target_enum {
@@ -781,6 +782,7 @@ struct TCP_Server_Info {
 	bool	sec_kerberosu2u;	/* supports U2U Kerberos */
 	bool	sec_kerberos;		/* supports plain Kerberos */
 	bool	sec_mskerberos;		/* supports legacy MS Kerberos */
+	bool	sec_iakerb;		/* supports pass-through auth for Kerberos (krb5 proxy) */
 	bool	large_buf;		/* is current buffer large? */
 	/* use SMBD connection instead of socket */
 	bool	rdma;
@@ -2148,6 +2150,8 @@ static inline char *get_security_type_str(enum securityEnum sectype)
 		return "Kerberos";
 	case NTLMv2:
 		return "NTLMv2";
+	case IAKerb:
+		return "IAKerb";
 	default:
 		return "Unknown";
 	}